src/zino/tasks/reachabletask.py

```diff
@@ -1,67 +1,75 @@
 import logging
-from datetime import datetime, timedelta
 
 from zino import state
 from zino.scheduler import get_scheduler
 from zino.snmp import SNMP
-from zino.statemodels import EventType
+from zino.statemodels import EventState, EventType, ReachabilityState
 from zino.tasks.task import Task
 
 _logger = logging.getLogger(__name__)
 
 
 class ReachableTask(Task):
-    EXTRA_JOBS_PREFIX = "delayed_reachable_job"
-    EXTRA_JOBS_INTERVALS = [60, 120, 240, 480, 960]
+    EXTRA_JOB_INTERVAL = 60
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._scheduler = get_scheduler()
 
     async def run(self):
-        """Checks if device is reachable. Schedules extra jobs if not."""
-        snmp = SNMP(self.device)
-        result = await snmp.get("SNMPv2-MIB", "sysUpTime", 0)
+        """Checks if device is reachable. Schedules extra reachability checks if not."""
+        if self._extra_job_is_running():
+            return
+        result = await self._get_sysuptime()
         if not result:
             _logger.debug("Device %s is not reachable", self.device.name)
             event, created = state.events.get_or_create_event(self.device.name, None, EventType.REACHABILITY)
             if created:
                 # TODO add attributes
-                event.add_log(f"{self.device.name} no-response")
-            if not self.extra_jobs_are_running():
-                self.schedule_extra_jobs()
+                event.state = EventState.OPEN
+                event.add_history("Change state to Open")
+            event.reachability = ReachabilityState.NORESPONSE
+            event.add_log(f"{self.device.name} no-response")
+            self._schedule_extra_job()
         else:
             _logger.debug("Device %s is reachable", self.device.name)
+
+    async def _run_extra_job(self):
+        result = await self._get_sysuptime()
+        if result:
+            _logger.debug("Device %s is reachable", self.device.name)
             event = state.events.get(self.device.name, None, EventType.REACHABILITY)
             if event:
                 # TODO update event attributes
+                event.reachability = ReachabilityState.REACHABLE
                 event.add_log(f"{self.device.name} reachable")
```

Comment:
Something of the same sort here: Only log that a device is reachable if its state switched from unreachable to reachable  (although, of course, this may already have that effect, since the extra job is descheduled when the device becomes available again)

Reply:
.. can add similar check here